src/ethereum/utils/numeric.py

```diff
@@ -165,3 +165,35 @@ def le_uint32_sequence_to_uint(sequence: Sequence[U32]) -> Uint:
     """
     sequence_as_bytes = le_uint32_sequence_to_bytes(sequence)
     return Uint.from_le_bytes(sequence_as_bytes)
+
+
+def fake_exponential(factor: int, numerator: int, denominator: int) -> int:
```

Comment:
Can we tighten the type bounds at all here? Like if all arguments are `Uint` the return value is also `Uint`? Should mean less casting downstream.

Reply:
Updated